xds/internal/balancer/clusterresolver/resource_resolver.go

```diff
@@ -30,16 +30,26 @@ import (
 // resourceUpdate is a combined update from all the resources, in the order of
 // priority. For example, it can be {EDS, EDS, DNS}.
 type resourceUpdate struct {
+	// A discovery mechanism would return an empty update when it runs into
+	// errors, and this would result in the priority LB policy reporting
+	// TRANSIENT_FAILURE (if there was a single discovery mechanism), or would
+	// fallback to the next highest priority that is available.
```

Comment:
In the not so very distant past, the `err` field in this struct was used to convey errors from the xDS client. This changed to be such that the individual resolvers would swallow connection errors and resource-not-found errors from the xDS client and would instead return no endpoints in this case. This change was done not only to be consistent with the other languages, but also to be able to continue using a previously good update (if one existed). But as part of that change, we failed to delete this `err` field. Now that I'm deleting it, I felt the need to document this behavior.

Did you find anything factually wrong with the comment? Or would you rather that this comment live somewhere else?

Reply:
I'm fine with it here thinking about it, was just wondering why you added it. At first I thought it's right (I know a priority's chain will report TF to trigger fallback) but I guess if there was one DM, this would trigger TF to be reported from priority. And if more, it would try to fallback to the next highest priority that is available (or TF the whole thing I guess).